src/admin/pages/posts/post.component.ts

```diff
@@ -35,10 +35,22 @@ export class PostComponent implements OnInit, OnDestroy {
 
     // File Upload
     private _imageToUpload = null;
+    private _imagePreviewContainer: HTMLElement;
 
     setFile({ target: { files } } = <any>{}) {
+        let imgPreviewContainer = this._imagePreviewContainer = this._document.getElementById('imagePreview');
         let [file] = files;
         this._imageToUpload = file;
+
+        let reader = new FileReader();
```

Comment:
while most modern browsers support FileReader, do we want a try catch around this line in case the current browser does not support it?  Just a thought...